doc/specs/#4999 - Improved keyboard handling in Conpty.md

```diff
@@ -0,0 +1,452 @@
+> NumLock key instead pressed, all the KeyState parameters would have bits 0x20
+> set. To get these keys with a NumLock, add 32 to the value.
+
+### Scenarios
+
+#### User is typing into WSL from the Windows Terminal
+
+`WT -> conpty[1] -> wsl`
+
+* Conpty[1] will ask for `win32-input-mode` from the Windows Terminal when
+  conpty[1] first boots up.
+* When the user types keys in Windows Terminal, WT will translate them into
+  win32 sequences and send them to conpty[1]
+* Conpty[1] will translate those win32 sequences into `INPUT_RECORD`s
+* When WSL reads the input, conpty[1] will translate the `INPUT_RECORD`s into VT
+  sequences corresponding to whatever input mode the linux app is in.
+
+#### User is typing into `cmd.exe` running in WSL interop
+
+`WT -> conpty[1] -> wsl -> conpty[2] -> cmd.exe`
+
+(presuming you start from the previous scenario, and launch `cmd.exe` inside wsl)
+
+* Conpty[2] will ask for `win32-input-mode` from conpty[1] when conpty[2] first
+  boots up.
+* When the user types keys in Windows Terminal, WT will translate them into
+  win32 sequences and send them to conpty[1]
+* Conpty[1] will translate those win32 sequences into `INPUT_RECORD`s
+* When WSL reads the input, conpty[1] will translate the `INPUT_RECORD`s into VT
+  sequences for the `win32-input-mode`, because it believes the client (in this
+  case, the conpty[2] running attached to `wsl`) wants `win32-input-mode`.
+* Conpty[2] will get those sequences, and will translate those win32 sequences
+  into `INPUT_RECORD`s
+* When `cmd.exe` reads the input, they'll receive the full `INPUT_RECORD`s
+  they're expecting
+
+
+## UI/UX Design
+
+This is not a user-facing feature.
+
+## Capabilities
+
+### Accessibility
+
+_(no change expected)_
+
+### Security
+
+_(no change expected)_
+
+### Reliability
+
+_(no change expected)_
+
+### Compatibility
+
+This isn't expected to break any existing scenarios. The information that we're
+passing to conpty from the Terminal should strictly have _more_ information in
+them than they used to. Conhost was already capable of translating
+`INPUT_RECORD`s back into VT sequences, so this should work the same as before.
+
+There's some hypothetical future where the Terminal isn't connected to conpty.
+In that future, the Terminal will still be able to work correctly, even with
+this ConPTY change. The Terminal will only switch into sending
+`win32-input-mode` sequences when _conpty asks for them_. Otherwise, the
+Terminal will still behave like a normal terminal emulator.
+
+### Performance, Power, and Efficiency
+
+_(no change expected)_
+
+## Potential Issues
+
+_(no change expected)_
+
+## Future considerations
+
+* We could also hypothetically use this same mechanism to send Win32-like mouse
+  events to conpty, since similar to VT keyboard events, VT mouse events don't
+  have the same fidelity that Win32 mouse events do.
+  - We could enable this with a different terminating character, like `"`

```

Comment:
I know this is only a future consideration, but note that `"` is not a valid final character. Maybe best to just leave out those details for now.

Reply:
Good catch, forgot that I put that line in there 😝 thanks!